File: src/344_reverse_string/reverse_string.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    void reverseString(vector<char>& s) {
        int left = 0, right = s.size() - 1;
        while (left < right) {
            swap(s[left], s[right]);
            left++;
            right--;
        }
    }
};

void printVec(const vector<char>& v) {
    cout << "[";
    for (size_t i = 0; i < v.size(); ++i) {
        cout << v[i];
        if (i + 1 < v.size()) cout << ", ";
    }
    cout << "]";
}

int main() {
    Solution sol;
    vector<char> arr1 = {'h','e','l','l','o'}; sol.reverseString(arr1); printVec(arr1); cout << " | [o, l, l, e, h]" << endl;
    vector<char> arr2 = {'a'}; sol.reverseString(arr2); printVec(arr2); cout << " | [a]" << endl;
    vector<char> arr3 = {}; sol.reverseString(arr3); printVec(arr3); cout << " | []" << endl;
    vector<char> arr4 = {'a','b'}; sol.reverseString(arr4); printVec(arr4); cout << " | [b, a]" << endl;
    vector<char> arr5 = {'r','a','c','e','c','a','r'}; sol.reverseString(arr5); printVec(arr5); cout << " | [r, a, c, e, c, a, r]" << endl;
    return 0;
}

File: src/344_reverse_string/ReverseString.java
import java.util.*;

public class ReverseString {
    public void reverseString(char[] s) {
        int left = 0, right = s.length - 1;
        while (left < right) {
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }
    }

    public static void main(String[] args) {
        ReverseString sol = new ReverseString();
        char[] arr1 = "hello".toCharArray();
        sol.reverseString(arr1);
        System.out.println(Arrays.toString(arr1) + " | [o, l, l, e, h]");
        char[] arr2 = "a".toCharArray();
        sol.reverseString(arr2);
        System.out.println(Arrays.toString(arr2) + " | [a]");
        char[] arr3 = "".toCharArray();
        sol.reverseString(arr3);
        System.out.println(Arrays.toString(arr3) + " | []");
        char[] arr4 = "ab".toCharArray();
        sol.reverseString(arr4);
        System.out.println(Arrays.toString(arr4) + " | [b, a]");
        char[] arr5 = "racecar".toCharArray();
        sol.reverseString(arr5);
        System.out.println(Arrays.toString(arr5) + " | [r, a, c, e, c, a, r]");
    }
}
